packages/react-native/android/src/main/java/backtrace/library/models/nativeHandler/CrashHandlerConfiguration.java

```diff
@@ -0,0 +1,89 @@
+package backtraceio.library.models.nativeHandler;
+
+import android.content.pm.ApplicationInfo;
+import android.text.TextUtils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import backtraceio.library.common.AbiHelper;
+import backtraceio.library.services.BacktraceCrashHandlerRunner;
+
+public class CrashHandlerConfiguration {
+
+    public static final String BACKTRACE_CRASH_HANDLER = "BACKTRACE_CRASH_HANDLER";
+    public static final Set<String> UNSUPPORTED_ABIS = new HashSet<String>(Arrays.asList(new String[]{"x86"}));
+    private static final String CRASHPAD_DIRECTORY_PATH = "/crashpad";
+
+    private static final String BACKTRACE_NATIVE_LIBRARY_NAME = "libbacktrace-native.so";
+
+
+    public Boolean isSupportedAbi() {
+        return isSupportedAbi(AbiHelper.getCurrentAbi());
+    }
+
+    public Boolean isSupportedAbi(String abi) {
+        return !this.UNSUPPORTED_ABIS.contains(abi);
+    }
+
+    public String getClassPath() {
+        return BacktraceCrashHandlerRunner.class.getCanonicalName();
+    }
+
+    public List<String> getCrashHandlerEnvironmentVariables(ApplicationInfo applicationInfo) {
+        return getCrashHandlerEnvironmentVariables(applicationInfo.sourceDir, applicationInfo.nativeLibraryDir, AbiHelper.getCurrentAbi());
+    }
+
+    public List<String> getCrashHandlerEnvironmentVariables(String apkPath, String nativeLibraryDirPath, String arch) {
+        final List<String> environmentVariables = new ArrayList<>();
+
+        // convert available in the system environment variables
+        for (Map.Entry<String, String> variable :
+                System.getenv().entrySet()) {
+            environmentVariables.add(String.format("%s=%s", variable.getKey(), variable.getValue()));
+        }
+        // extend system-specific environment variables, with variables needed to properly run app_process via crashpad
+        File nativeLibraryDirectory = new File(nativeLibraryDirPath);
+
+        String backtraceNativeLibraryPath = getBacktraceNativeLibraryPath(nativeLibraryDirPath, apkPath, arch);
+        File allNativeLibrariesDirectory = nativeLibraryDirectory.getParentFile();
+        String allPossibleLibrarySearchPaths = TextUtils.join(File.pathSeparator, new String[]{
+                nativeLibraryDirPath,
+                allNativeLibrariesDirectory.getPath(),
+                System.getProperty("java.library.path"),
+                "/data/local"});
```

Comment:
won't this get joined as `...path//data/local`?

Reply:
no, it will be separated by `:`

output

```
/data/app/~~nlc7R6EdprMECkAbt0jRFw==/com.reactnative-u_yqZOmeMUdl3wdqYW_3_g==/lib/arm64:/data/app/~~nlc7R6EdprMECkAbt0jRFw==/com.reactnative-u_yqZOmeMUdl3wdqYW_3_g==/lib:/system/lib64:/system_ext/lib64:/data/local
```